### PR TITLE
Try setting Java version correctly via. substitution

### PR DESCRIPTION
**Pull request by dzou**
This should allow GraalVM images to correctly set the Java language header to specify usage of the GraalVM distribution of Java over standard Java.

### CODE REVIEW

File: google-cloud-graalvm-support/src/main/java/com/google/cloud/graalvm/features/substitutions/GaxPropertiesSubstitutions.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * This file contains the GaxProperties substitution to correctly set the Java language string
+ * in API call headers for GraalVM users.
+ */
+class GaxPropertiesSubstitutions {
+}
+
+@TargetClass(className = "com.google.api.gax.core.GaxProperties")
+final class Target_com_google_api_gax_core_GaxProperties {
```
Comment (meltsufin) [inline google-cloud-graalvm-support/src/main/java/com/google/cloud/graalvm/features/substitutions/GaxPropertiesSubstitutions.java:30]: Does the class name have to look like that? It seems redundant given the annotation specifying the target class.

**Review by meltsufin — APPROVED** [review]

File: google-cloud-graalvm-support/src/main/java/com/google/cloud/graalvm/features/substitutions/GaxPropertiesSubstitutions.java (hunk 1 above)
Comment (dzou) [reply google-cloud-graalvm-support/src/main/java/com/google/cloud/graalvm/features/substitutions/GaxPropertiesSubstitutions.java]: Done - restored to simpler style. Yeah I copied the style from another repo, but then doing some more studying some more examples there appears to be no established convention so I'll keep it simple.

File: src/checkstyle/suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<!DOCTYPE suppressions PUBLIC
+  "-//Checkstyle//DTD SuppressionFilter Configuration 1.2//EN"
+  "https://checkstyle.org/dtds/suppressions_1_2.dtd">
+<suppressions>
+  <suppress files="com[\\/]google[\\/]cloud[\\/]graalvm[\\/]features[\\/]substitutions[\\/]" checks="TopLevelClass|TypeName" />
```
Comment (elefeint) [inline src/checkstyle/suppressions.xml:6]: You probably don't need this anymore after changing the class name.

**Review by elefeint — CHANGES_REQUESTED** [review]

File: src/checkstyle/suppressions.xml (hunk 2 above)
Comment (dzou) [reply src/checkstyle/suppressions.xml]: Ah good catch - removed.

**Review by elefeint — APPROVED** [review]

**Pull request merged**